.github/workflows/pr-lint.yaml

```diff
@@ -126,3 +126,32 @@ jobs:
           fi
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  validate-dependency-mismatches:
+    name: Check Dependency Mismatches
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: CI Setup
+        uses: ./.github/actions/ci-setup
+
+      - name: Install Dependencies
+        run: pnpm install
+
+      - name: Install syncpack
+        run: pnpm add --save-dev syncpack
```

Comment:
Better to have this as a development dependency on the monorepo, so we can version it properly. And we have the config file committed anyways.